Support the ability to exclude specific font types

mike183 opened the pull request:
This PR introduces a new option named `fontFilePattern`, that allows the user to configure which font types they would like to be included in the build output.

Usage is based on the [filePattern](https://github.com/ember-cli-deploy/ember-cli-deploy-s3#filepattern) option used in [ember-cli-deploy-s3](https://github.com/ember-cli-deploy/ember-cli-deploy-s3).

The default value for this option is `*.{eot,svg,ttf,woff,woff2,otf}`, which includes all of the font types supplied by FontAwesome, meaning that the change should be fully backwards compatible.

martndemus commented on index.js:
```diff
@@ -13,9 +13,12 @@ module.exports = {
 
   treeForVendor: function(tree) {
+    // Get configured fontFilePattern
+    let fontFilePattern = this.options.fontFilePattern || "*.{eot,svg,ttf,woff,woff2,otf}";
```
Could you make this something that accepts an array of file types instead of a hacky string?
Then you can `*.{${fontFilePatterns.join(',')}}`;

mike183 commented:
Required font formats are now supplied using an array as requested (much nicer API).

I also fixed a couple of JSHint errors and removed the `brocolli-merge-trees` dependency which wasn't being used.
